docs/product/components/modals.html

```diff
@@ -52,24 +37,88 @@ <h1 class="s-modal--header" id="modal-title">Example Danger Title</h1>
 <section class="stacks-section">
     {% header h2 | Examples %}
     <div class="mb24 p16 bar-sm bg-black-025 ba bc-black-1 grid gs8 fw-wrap">
-        <button class="grid--cell s-btn s-btn__filled js-modal-open" data-toggle="s-modal" data-target="#modal-base">Launch example modal</button>
-        <button class="grid--cell s-btn s-btn__filled js-modal-open" data-toggle="s-modal" data-target="#modal-danger">Launch example modal w/ danger state</button>
+
+        <div class="grid--cell" data-controller="s-modal">
+            <button class="s-btn s-btn__filled" data-action="s-modal#show">Launch example modal</button>
+            <aside
+                class="s-modal"
+                tabindex="-1"
+                role="dialog"
+                aria-labelledby="modal-title"
+                aria-describedby="modal-description"
+                aria-hidden="true"
+                data-target="s-modal.content"
+                data-action="click->s-modal#hide"
+            >
+                <div class="s-modal--dialog" role="document" data-action="click->s-modal#nohide">
+                    <h1 class="s-modal--header" id="modal-title">Example title</h1>
+                    <p class="s-modal--body" id="modal-description">Nullam ornare lectus vitae lacus sagittis, at sodales leo viverra. Suspendisse nec nulla dignissim elit varius tempus. Cras viverra neque at imperdiet vehicula. Curabitur condimentum id dolor vitae ultrices. Pellentesque scelerisque nunc sit amet leo fringilla bibendum. Etiam feugiat imperdiet mi, eu blandit arcu cursus a.</p>
+                    <div class="grid gs8 gsx s-modal--footer">
+                        <button class="grid--cell s-btn s-btn__primary" type="button" data-action="s-modal#hide">Save changes</button>
+                        <button class="grid--cell s-btn" type="button" data-action="s-modal#hide">Cancel</button>
+                    </div>
+                    <a class="s-modal--close s-btn s-btn__muted" href="#" aria-label="Close" data-action="s-modal#hide">
+                        {% icon ClearSm %}
+                    </a>
+                </div>
+            </aside>
+        </div>
+
+        <div class="grid--cell" data-controller="s-modal">
+            <button class="s-btn s-btn__filled" data-action="s-modal#show">Launch example modal w/ danger state</button>
+            <aside
```

Comment:
This whitespace pattern is so weird to me, but I understand why we do it. I might format this slightly differently.

Reply:
No strong opinions re formatting, I just found this easier to read than having one massive line. Knock yourself out 😉